### ISSUE TITLE
Add support for the UI block in task groups, tasks, and allocations

### ISSUE THREAD

**Issue by josegonzalez** [hashicorp/nomad#24103]
### Proposal

Add support for the `ui` block to be defined in `job.group` and `job.group.task`. Additionally, add support for alloc-specific ui links. Both could be achievable via some sort of url templating schema:

```hcl
# replacements
# - ${job_name}
# - ${group_name}
# - ${task_name}
# - ${allocation_id}

job "vector" {
  group "vector" {
    task "vector" {
      ui {
        link {
          allocation_link = true
          label = "logs"
          url = "https://logs-ui.example.com/search?job=${job_name}&group=${group_name}&task=${task_name}&alloc=${allocation_id}"
        }
      }
    }
  }
}
```

### Use-cases

For more complex jobs, it may be useful to link out to external services - logs and metrics being the key ones - at a group or task level. This would allow for aggregated data that is specific to those levels - which is important in jobs that have multiple groups, or cases where there are multiple tasks within a group (such as for sidecars).

Additionally, having links out for _just_ allocations would allow folks to filter to the exact allocation that is being looked at. This is super useful when trying to figure out why one or more allocations are misbehaving.

### Attempted Solutions

At the moment, I need to copy the allocation ID and then paste it into a form elsewhere. This works for me but developers may be less clear as to where to browse on external services to filter data. Worse still, it makes it more difficult to migrate across external services (as in, if I move from Grafana => DataDog or back for logs/metrics) as now I need to communicate that change to developers and teach them the new way to do things. Having those links be embeddable in the Nomad UI will make it easier for folks to _use_ those links.

**Comment by philrenaud** [comment]
Hi @josegonzalez , thanks for this feature request. I think it's a good one; my only pause is around the `allocation_link` field and I'd like to understand it better.

Is the idea that with `allocation_link` set to `true`, the link would show up at `/ui/allocations/$allocID` (potentially several of them, if each of an allocation's tasks had a ui link)? Or on `/ui/allocations/$allocID/$taskName` with some tag or icon indicating it's allocation-specific?

**Comment by Kamilcuk** [comment]
Consider:

```
job "vector" {
  group "vector" {
    task "vector" {
      meta {
        link_to_your_logs = "https://logs-ui.example.com/search?job=${NOMAD_JOB_NAME}&group=${NOMAD_GROUP_NAME}&task=${NOMAD_TAKS_NAME}&alloc=${NOMAD_ALLOC_ID}"
```

Then you can copy the meta from ui into browser.

**Comment by josegonzalez** [comment]
Yeah I wonder if there is a difference between an alloc and a task. I think it might be good to have something that is global to the alloc, and then another that is task-specific. Not sure what that would look like in terms of the HCL, but good callout.

**Comment by enspritz** [comment]
We need exactly this. 

```hcl
job "..." {
  group "..." {
    count = n
    task "service" {
      ui {
        link {
          label = "${NOMAD_IP_admin_ui} admin UI"
          url = "${NOMAD_ADDR_admin_ui}"
        }
   ...
}
```
Where the group configured with `count = 5`, 5 buttons would appear on the Nomad job page. And if I can wish, 5 buttons on the task group page, and a single button on each task page. Each provides an easy jumping-off point to hit the admin pages of any instance appropriate to the Nomad view/page.
